easybuild/tools/filetools.py

```diff
@@ -2980,3 +2987,81 @@ def create_unused_dir(parent_folder, name):
     set_gid_sticky_bits(path, recursive=True)
 
     return path
+
+
+def get_first_nonexisting_parent(path):
+    path = os.path.abspath(path)
+
+    first_nonexisting_parent = None
+    while not os.path.exists(path):
+        first_nonexisting_parent = path
+        path = os.path.dirname(path)
+
+    return first_nonexisting_parent
+
+
+def _apply_suffix(paths, number):
+    suffix = '_' + str(number)
+
+    final_paths = []
+    first_nonexisting_parents = []
+    for p in paths:
+        final_paths.append(p + suffix)
+        first_nonexisting_parents.append(p)
+
+    return final_paths, first_nonexisting_parents
+
+
+def _clean_directories(paths):
+    for p in paths:
+        shutil.rmtree(p, ignore_errors=True)
+
+
+def create_unused_dirs(paths, index_upper_bound=10000):
+    """
+    Create directories with given paths, including the parent directories
+    When a directory in the same path for any of the path  in the path list already exists, then the suffix '_<i>' is
+    appended for i=0..(index_upper_bound-1) until an index is found so that all required path as unused. All created
+    directories have the same postfix.
+
+    :param paths: list of paths where directories are created; parent path created as well
+    :param index_upper_bound: maximum index that will be tried before failing
+    """
+    paths = list(map(os.path.abspath, paths))
+    for path in paths:
+        for parent in set(paths) - {path}:
+            if is_predecessor(parent, path):
+                raise EasyBuildError("Path '%s' is predecesor of '%s'" % (parent, path))
+
+    final_paths = paths
+    first_nonexisting_parents = list(map(get_first_nonexisting_parent, paths))
+    paths_exist = False
+    number = -1
+    while number < index_upper_bound and not paths_exist:  # Start with no suffix and limit the number of attempts
```

Comment:
What about 
```suggestion
    while for number in range(-1, 10000):  # Start with no suffix and limit the number of attempts
```

Is the argument to the function for the upper bound really required? With this you can instead of `if number == index_upper_bound and not paths_exist:` just use a flag `valid_directories_created` or so and `break` in the loop

Reply:
I try to avoid `break`. Is it this bad to leave the while loop?

The index upper bound does not have to be exposed for most reasonable use cases. It is convenient however for testing.

Reply:
It's probably up to personal preference. My line of though is:
- We have essentially a counting loop, so `for` is more suitable
- `for` avoids having some code path accidentally skip the counter-increment leading to an infinite loop in some corner case
- in other languages it avoids "leaking" the counter variable to outside scope, not in Python though
- `x = False; while not x` looks a bit odd, but that is Python not having do-while loops
- My logic is `for all attempts: try-run_task; if task-succeeded: stop looping`

Maybe some middle ground if you don't want the break in a "hidden place":
```
for number in range(-1, 10000):
    if paths_exist:
        break
```
This avoids the pitfall of missing the increment and should still read similar to `while ... not paths_exist` but split into 2 logical blocks.

Many ideas, all work, you (or in the end the maintainers) decide.

I added a suggestion for your current version which to me is a bit more clear. E.g. `if cond: x = y else: x=z` and moving `tried_paths = []` right before it is filled to make it easier to see that it starts of as empty. You also don't need to check `number` after the loop as `not paths_exist` is only met when the number of tries are exceeded.